ci/cloudrunners/oci/main.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/cncf/automation/cloudrunners/oci/pkg/oci"
+	"github.com/cncf/automation/cloudrunners/pkg/remote"
+	"github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/oracle/oci-go-sdk/v65/core"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh"
+)
+
+var Cmd = &cobra.Command{
+	Use:  "gha-runner",
+	Long: "Run a GitHub Actions runner (on Oracle Cloud Infrastructure)",
+	RunE: run,
+}
+var args struct {
+	debug bool
+
+	arch               string
+	compartmentId      string
+	subnetId           string
+	availabilityDomain string
+}
+
+func main() {
+	log.SetFlags(log.Flags() | log.Lshortfile)
+
+	if err := Cmd.Execute(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+
+	os.Exit(0)
+}
+
+func run(cmd *cobra.Command, argv []string) error {
+	ctx := context.Background()
+	// Initialize the OCI client
+	computeClient, err := core.NewComputeClientWithConfigurationProvider(common.DefaultConfigProvider())
+	if err != nil {
+		return fmt.Errorf("failed to create compute client: %w", err)
+	}
+	networkClient, err := core.NewVirtualNetworkClientWithConfigurationProvider(common.DefaultConfigProvider())
+	if err != nil {
+		return fmt.Errorf("failed to create network client: %w", err)
+	}
+
+	// List Images and retrieve the latest ID by type and arch
+
+	images, err := computeClient.ListImages(ctx, core.ListImagesRequest{
+		CompartmentId:   common.String("ocid1.compartment.oc1..aaaaaaaa22icap66vxktktubjlhf6oxvfhev6n7udgje2chahyrtq65ga63a"),
+		OperatingSystem: common.String("ubuntu-24.04-x86-gha-image"),
+		SortBy:          core.ListImagesSortByTimecreated,
+		SortOrder:       core.ListImagesSortOrderDesc,
+		Limit:           common.Int(1),
+	})
+	if err != nil {
+		panic(err)
+	}
+	if len(images.Items) == 0 {
+		return fmt.Errorf("no images found")
+	}
+	latestImage := images.Items[0]
+
+	// Create SSH Key Pair
+	sshKeyPair, err := remote.CreateSSHKeyPair()
+	if err != nil {
+		return fmt.Errorf("creating ssh key pair: %w", err)
+	}
+
+	sshKeys := "gha:" + sshKeyPair.PublicKey
```

Comment:
This "gha:" part does not work. Maybe remove this line. I will add an update for the metadata part